frappe/custom/doctype/customize_form/customize_form.py

```diff
@@ -195,72 +152,142 @@ def save_customization(self):
 
 	def set_property_setters(self):
 		meta = frappe.get_meta(self.doc_type)
-		# doctype property setters
 
-		for property in doctype_properties:
-			if self.get(property) != meta.get(property):
-				self.make_property_setter(property=property, value=self.get(property),
-					property_type=doctype_properties[property])
+		# doctype
+		self.set_property_setters_for_doctype(meta)
 
+		# docfield
 		for df in self.get("fields"):
 			meta_df = meta.get("fields", {"fieldname": df.fieldname})
-
 			if not meta_df or meta_df[0].get("is_custom_field"):
 				continue
+			self.set_property_setters_for_docfield(meta, df, meta_df)
+
+		# action and links
+		self.set_property_setters_for_actions_and_links(meta)
+
+	def set_property_setters_for_doctype(self, meta):
+		for prop, prop_type in doctype_properties.items():
+			if self.get(prop) != meta.get(prop):
+				self.make_property_setter(prop, self.get(prop), prop_type)
+
+	def set_property_setters_for_docfield(self, meta, df, meta_df):
+		for prop, prop_type in docfield_properties.items():
+			if prop != "idx" and (df.get(prop) or '') != (meta_df[0].get(prop) or ''):
+				if not self.allow_property_change(prop, meta_df, df):
+					continue
+
+				self.make_property_setter(prop, df.get(prop), prop_type,
+					fieldname=df.fieldname)
+
+	def allow_property_change(self, prop, meta_df, df):
+		if prop == "fieldtype":
+			self.validate_fieldtype_change(df, meta_df[0].get(prop), df.get(prop))
+
+		elif prop == "allow_on_submit" and df.get(prop):
+			if not frappe.db.get_value("DocField",
+				{"parent": self.doc_type, "fieldname": df.fieldname}, "allow_on_submit"):
+				frappe.msgprint(_("Row {0}: Not allowed to enable Allow on Submit for standard fields")\
+					.format(df.idx))
+				return False
+
+		elif prop == "reqd" and \
+			((frappe.db.get_value("DocField",
+				{"parent":self.doc_type,"fieldname":df.fieldname}, "reqd") == 1) \
+				and (df.get(prop) == 0)):
+			frappe.msgprint(_("Row {0}: Not allowed to disable Mandatory for standard fields")\
+					.format(df.idx))
+			return False
+
+		elif prop == "in_list_view" and df.get(prop) \
+			and df.fieldtype!="Attach Image" and df.fieldtype in no_value_fields:
+					frappe.msgprint(_("'In List View' not allowed for type {0} in row {1}")
+						.format(df.fieldtype, df.idx))
+					return False
+
+		elif prop == "precision" and cint(df.get("precision")) > 6 \
+				and cint(df.get("precision")) > cint(meta_df[0].get("precision")):
+			self.flags.update_db = True
+
+		elif prop == "unique":
+			self.flags.update_db = True
+
+		elif (prop == "read_only" and cint(df.get("read_only"))==0
+				and frappe.db.get_value("DocField", {"parent": self.doc_type,
+				"fieldname": df.fieldname}, "read_only")==1):
+			# if docfield has read_only checked and user is trying to make it editable, don't allow it
+			frappe.msgprint(_("You cannot unset 'Read Only' for field {0}").format(df.label))
```

Comment:
```suggestion
			frappe.msgprint(_("You cannot unset 'Read Only' for field {0}").format(df.label), title=_("Validation Error"))
```
Can we have a `title` for each validation `msgprint`? Or else it looks as follows.
<img width="437" alt="Screenshot 2020-10-22 at 11 17 57 AM" src="https://user-images.githubusercontent.com/13928957/96831910-7efcac80-145b-11eb-9d93-6fe67a14b44e.png">